scripts/buildorder.py

```diff
@@ -347,19 +315,49 @@ def generate_target_buildorder(target_path, pkgs_map, fast_build_mode):
         package.deps.difference_update([subpkg.name for subpkg in package.subpkgs])
     return package.recursive_dependencies(pkgs_map)
 
+def get_list_cyclic_dependencies(pkgs_map, index=None, checked=None, pkgname=None, return_result=False):
```

Comment:
The `return_result` variable means the return of a variable such as `result`, which is a list in the `get_list_cyclic_dependencies` function.  When set to `False`, it does not return this value.

> Make a function that evaluates cyclic dependencies and returns them; in the places where you want to print that state, print it there (or call another function that prints the result this function returns).

I think I'll leave printing cyclic dependencies inside the function, since this method shortens the code.